docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,471 @@
+Then upload it to a distributed storage.
+```bash
+# Upload to HDFS
+hdfs dfs -put /path/to/local/data/FashionMNIST hdfs://url:port/path/to/remote/data
+```
+
+# 4. Prepare Custom Modules
+Spark allows to upload Python files(`.py`), and zipped Python packages(`.zip`) to the executors by setting `--py-files` option in Spark scripts or `extra_python_lib` in `init_orca_context`. 
+
+The example depends on few Python files, you could upload them to YARN on __Client Node__ (where you submit the program to YARN) directly.
+* When using `python` command, please specify `extra_python_lib` in `init_orca_context`.
+    ```python
+    init_orca_context(cluster_mode, cores, memory, num_nodes,
+                      extra_python_lib="/path/to/model.py")
+    ```
+
+    Please see more details in [OrcaContext Python Dependencies](https://bigdl.readthedocs.io/en/latest/doc/Orca/Overview/orca-context.html#python-dependencies).
+
+* When using `bigdl-submit` or `spark-submit` script, please specify `--py-files` option.
+    ```bash
+    --py-files /path/to/model.py
+    ```
+
+    Please see more details in [Spark Submitting Applications](https://spark.apache.org/docs/latest/submitting-applications.html). 
+
+## Optional (Use Zipped Packages)
+If your program depends on a nested directory of Python files, you could package them into a zipped packages on the __Client Node__ (where you submit applications) instead. 
+
+For example, `example` is the nested directory including `model.py` needed for this program.
+
+```bash
+zip -q -r orca_example.zip example
+```
+
+__Note:__
+* Please follow the previous part to upload zipped Python packages to cluster.
+* You should import custom modules from the unzipped file as below:
+    ```python
+    from example.model import model_creator, optimizer_creator
+    ```
+
+# 5. Run Jobs on YARN
+In the following part, you will learn three ways that BigDL supports to submit and run the example on YARN:
+* Use `python` command
+* Use `bigdl-submit`
+* Use `spark-submit`
+
+## 5.1 Use `python` Command
+This is the easiest and most recommended way to run BigDL on YARN.
+
+__Note:__
+* You only need to prepare the environment on the __Client Node__(where you submit applications), all dependencies would be automatically packaged and distributed to YARN cluster.
+
+### 5.1.1 Yarn Client
+Please call `init_orca_context` to create an OrcaContext at the very beginning of each Orca program.
+```python
+from bigdl.orca import init_orca_context
+
+init_orca_context(cluster_mode="yarn-client", cores=4, memory="10g", num_nodes=2, 
+                  driver_cores=2, driver_memory="4g",
+                  extra_python_lib="/path/to/model.py")
+``` 
+Run the example following command below:
+```bash
+python train.py --cluster_mode yarn-client --remote_dir hdfs://url:port/path/to/remote/data
+```
+* `--cluster_mode`: set the cluster_mode in `init_orca_context`.
+* `--remote_dir`: load dataset and save model directory on a distributed storage.
```

Comment:
directory on a distributed storage for the dataset (see section 3) and saving the model